### ISSUE TITLE
mjpeg streaming from net (through named pipe) does not always work

### ISSUE THREAD

**Issue by shuckle50** [huceke/omxplayer#162]
I am trying to see the my web camera feeds with omxplayer. My cameras only send video (no sound).
The problem is that very often omxplayer does not show anything on the screen. It replies fast to any key commands though and for example if I pause it (with p), I see that wget also stops writing into the named pipe, so it very much looks like omxplayer is alive. It just does not display anything to the screen.
But sometimes it works fine. 

I get the data with wget from local zoneminder
wget -O /run/stream.mjpeg 'http://192.168.254.212/cgi-bin/nph-zms?monitor=2&maxfps=10&buffer=4000'
and then read it with omxplayer in another terminal session:
 omxplayer  --win "640 544 1280 1024"  /run/stream.mjpeg

I have also tried without the --win parameter and I tested with several --video_queue and --video_fifo settings (like 5 and 10) but I did not see any effects form those.

I wrote three examples from wget also to files and they all work in my linux laptop with mplayer (wrong speed, but picture is visible). But only one of the examples work with omxplayer.

Sorry, forgot to add my versions:
omxplayer_0.2.5~git20130326~5b369233_armhf.deb

 $ uname -a
Linux raspberry4 3.6.11+ #414 PREEMPT Thu Apr 18 02:00:59 BST 2013 armv6l GNU/Linux

$ vcgencmd version
Apr 18 2013 02:23:00 
Copyright (c) 2012 Broadcom
version 384972 (release)

Any ideas what to test?

**Comment by aquadat0r** [comment]
Yes, seems like similar issue that I raised https://github.com/huceke/omxplayer/issues/160

**[works]** omxplayer_0.2.5~git20130326~5b369233_armhf.deb (41ac5e9)
**[don't work]** omxplayer_0.2.5~git20130421~ae4dc6e9_armhf.deb (dd04552)

Although it seems you are having issues with the release that works for me. Can you double check your omxplayer build? or perhaps uninstall and reinstall that specific release and try again?

**Comment by shuckle50** [comment]
I tested with versions:

omxplayer_0.2.5~git20130326~5b369233_armhf.deb
omxplayer_0.2.5~git20130421~ae4dc6e9_armhf.deb
omxplayer_0.2.6~git20130427~fcfb7911_armhf.deb

and the results were exactly the same.

Is there a way to download the test files to this issue report?

**Comment by shuckle50** [comment]
I found a way to do this. I need to pipe the output through avconv and it becomes readable by omxplayer.
Currently I seem to get good results by running this script in my "mainframe" big machine (ip 192.168.0.104):

mkfifo /tmp/stream.mjpeg /tmp/avconv.mjpeg
wget -O /tmp/stream.mjpeg 'http://192.168.254.212/cgi-bin/nph-zms?monitor=2&maxfps=10&buffer=4000'&
sleep 1
avconv -y -i /tmp/stream.mjpeg -same_quant /tmp/avconv.mjpeg&
sleep 1
pv -b /tmp/avconv.mjpeg | nc -l 34567

and then read and show it in my small Raspberry Pi:
Session1:
sudo mkfifo /run/stream.mjpeg
sudo chmod 777 /run/stream.mjpeg
nc 192.168.0.104 34567 > /run/stream.mjpeg

Session2:
omxplayer  --win "640 544 1280 1024"  /run/stream.mjpeg

This seems to work almost always and stays up long. I will test how long.
I am using 5b369233 version because I run into https://github.com/huceke/omxplayer/issues/160 with versions ae4dc6e9 and fcfb7911. (it works, but I need to pause and stop it before it becomes visible.)

Now if only I could run more than one mjpeg stream simultaneously.

**Comment by shuckle50** [comment]
I tested this with the new omxplayer and it now says:

$ omxplayer  --info --stats avconv2.mjpeg 
[mjpeg @ 0x22b5a0] Could not find codec parameters for stream 0 (Video: mjpeg): unspecified size
Consider increasing the value for the 'analyzeduration' and 'probesize' options
[mjpeg @ 0x22b5a0] Estimating duration from bitrate, this may be inaccurate

have a nice day ;)

Any ideas?

$ omxplayer  --ver
omxplayer - Commandline multimedia player for the Raspberry Pi
        Build date: Tue, 18 Jun 2013 14:05:41 +0200
        Version   : 8486405 [master]
        Repository: https://github.com/popcornmix/omxplayer.git

**Comment by popcornmix** [comment]
```
$ omxplayer --info --stats avconv2.mjpeg
[mjpeg @ 0x22b5a0] Could not find codec parameters for stream 0 (Video: mjpeg): unspecified size
Consider increasing the value for the 'analyzeduration' and 'probesize' options
```

This is ffmpeg code before it's even got to omxplayer, so I'm not sure what we can do about that.
Does this scheme work with ffmpeg (version 1.2) rather than omxplayer?
(on Pi or other platform).

**Comment by shuckle50** [comment]
I guess this can't be solved then. 
I do not know how to install ffmpeg version 1.2 nor do I know how to test with ffmpeg.

I really would have liked to utilize the GPU to show these two mjpeg streams as the CPU is really not able to do it while the GPU is just sleeping. But I have failed to convert those web camera streams to any format that omxplayer would be able to play. So sad that utlising the GPU is so hard.

**Comment by Grunge** [comment]
same here:

with version 0.3.0:

root@raspberrypi:~# omxplayer -s -i --win "0 0 960 720" rtsp://--some-ip--/mpeg4/media.amp
[rtsp @ 0xa02f70] Could not find codec parameters for stream 0 (Video: mpeg4): unspecified size
Consider increasing the value for the 'analyzeduration' and 'probesize' options

have a nice day ;)

with version 0.2.6 downgrade its running again:

omxplayer -s -i --win "0 0 960 720" rtsp://46.234.32.81/mpeg4/media.amp
[rtsp @ 0x3ecbf0] Estimating duration from bitrate, this may be inaccurate
Input #0, rtsp, from 'rtsp://46.234.32.81/mpeg4/media.amp':
  Metadata:
    title           : Media Presentation
  Duration: N/A, start: 0.173889, bitrate: N/A
    Stream #0:0: Video: mpeg4 (Advanced Simple Profile), yuv420p, 640x480 [SAR 1:1 DAR 4:3], 100 tbr, 90k tbn, 100 tbc

have a nice day ;)

Way to downgrade:

wget http://omxplayer.sconde.net/builds/omxplayer_0.2.6~git20130427~fcfb7911_armhf.deb
dpkg -i omxplayer_0.2.6~git20130427~fcfb7911_armhf.deb 

Greetings

**Comment by Grunge** [comment]
i had also probs with 0.2.6 and i'm using now, successful version 0.2.7

omxplayer_0.2.7~git20130622~af28ef87_armhf.deb

wget http://omxplayer.sconde.net/builds/omxplayer_0.2.7~git20130622~af28ef87_armhf.deb
dpkg -i omxplayer_0.2.7~git20130622~af28ef87_armhf.deb

Greetings

**Comment by popcornmix** [comment]
@shuckle50 is this still a problem with latest omxplayer deb?

**Comment by shuckle50** [comment]
No luck for me.
Still only omxplayer_0.2.5~git20130326~5b369233_armhf.deb works for me.
I built another test file and running with the mentioned old version gives:

$ omxplayer -r test.mjpeg 
Video codec omx-mjpeg width 640 height 480 profile -99 fps 25.000000
Subtitle count: 0, state: off, index: 1, delay: 0

have a nice day ;)

$ omxplayer -info test.mjpeg 
[mjpeg @ 0xab2610] Estimating duration from bitrate, this may be inaccurate
Input #0, mjpeg, from 'testi.mjpeg':
  Duration: N/A, bitrate: N/A
    Stream #0:0: Video: mjpeg, yuvj420p, 640x480 [SAR 1:1 DAR 4:3], 25 fps, 25 tbr, 1200k tbn, 25 tbc

have a nice day ;)

While the new one ends immediately with:
$ omxplayer -r test.mjpeg 
have a nice day ;)

$ omxplayer -info test.mjpeg 
[mjpeg @ 0x14595e0] Could not find codec parameters for stream 0 (Video: mjpeg): unspecified size
Consider increasing the value for the 'analyzeduration' and 'probesize' options
[mjpeg @ 0x14595e0] Estimating duration from bitrate, this may be inaccurate
have a nice day ;)

 $ omxplayer -v
omxplayer - Commandline multimedia player for the Raspberry Pi
        Build date: Sat, 22 Jun 2013 22:05:49 +0200
        Version   : af28ef8 [master]
        Repository: https://github.com/popcornmix/omxplayer.git

**Comment by popcornmix** [comment]
@shuckle50 
Can you post a link to a sample file that fails?

**Comment by shuckle50** [comment]
These three are small test files. The number is fps. They all can be played with the old omxplayer, but not with the new one. Test with mplayer -fps n testn.mjpeg.
http://www.kolumbus.fi/kautto.juha/test10.mjpeg
http://www.kolumbus.fi/kautto.juha/test2.mjpeg
http://www.kolumbus.fi/kautto.juha/test4.mjpeg

This is similar, but bigger files, so that you have time to see it properly (as those are played with wrong speed).
http://www.kolumbus.fi/kautto.juha/testi.mjpeg

It would actually be nice if omxplayer had fps parameter also like mplayer.

**Comment by shuckle50** [comment]
The last version which works for me is:
omxplayer_0.2.6~git20130427~fcfb7911_armhf.deb

And the first version, which I found not working is:
omxplayer_0.2.7~git20130618~8486405a_armhf.deb

I am not aware of any versions between those, so I have not tested them.

I just tested also:
omxplayer_0.3.0~git20130815~c99bc80_armhf.deb
and that did not work. 

The size of the working versions is about 11M while the new versions are smaller (about 8M)

**Comment by shuckle50** [comment]
I tested this more and after doing rpi-update this now works perfectly. I can run two streams at the same time and it is rock solid.

Improvements after rpi-update:
1) It has never failed, when before it failed about 50% of cases. About 20 trials.
2) It seems to run forever. I have run two streams more then 24 hours without issues while before it only lasted max few hours.
3) I do not have to press pause to get the streaming started. Before I had to press space bar two times quickly enough to get it working.

Now my only issue is that new versions of omxplayer do not work.
